### PR TITLE
#79: Implemente first accessibilityIdentifier-based UI test

### PR DESCRIPTION
**Pull request by artdaw**
#79: Implemented first accessibilityIdentifier-based UI test

### CODE REVIEW

**Comment by bonkey** [comment]
It reads for me better if accesibilityIdentifier doesn't have class name split with dash, so instead `catalog-navigation-controller`, rather `catalog-navigationcontroller`, so there's "function-type" syntax. Or maybe we could drop class names from there completely? Not strongly for/against it, however.

File: AtlasDemo/AtlasDemoUITests/AtlasDemoUITests.swift [hunk 1]
```diff
@@ -28,40 +28,36 @@ class AtlasDemoUITests: XCTestCase {
...
 
         tapConnectAndLogin()
-        tapBuyNow()
+        tapPlaceOrder()
         tapBackToShop()
     }
 
     func testDeleteAddress() {
-        let sizeText = app.tables.staticTexts["42"]
+        let size = app.cells["size-cell-1"]
```
Comment (bonkey) [inline AtlasDemo/AtlasDemoUITests/AtlasDemoUITests.swift]: Let's pull it inline into `waitForElementToAppearAndTap` call

File: AtlasDemo/AtlasDemoUITests/AtlasDemoUITests.swift [hunk 2]
```diff
@@ -28,40 +28,36 @@ class AtlasDemoUITests: XCTestCase {
...
         tapBuyNow("Lamica")
-        waitForElementToAppearAndTap(sizeText)
+        waitForElementToAppearAndTap(size)
         tapConnectAndLogin()
-        app.scrollViews.childrenMatchingType(.Any)
-            .element.childrenMatchingType(.Any).elementBoundByIndex(2).staticTexts["Erika Mustermann, Mollstr. 1 10178 Berlin"].tap()
+        app.otherElements["shipping-stack-view"].tap()
         deleteAddresses()
-        app.navigationBars["Summary"].buttons["Cancel"].tap()
+        app.buttons["navigation-bar-cancel-button"].tap()
```
Comment (bonkey) [inline AtlasDemo/AtlasDemoUITests/AtlasDemoUITests.swift]: Maybe "navigation.cancel" name?

Comment (bonkey) [inline AtlasDemo/AtlasDemoUITests/AtlasDemoUITests.swift]: Or rather use `.navigationBars` query and just `cancel`?

File: AtlasDemo/AtlasDemoUITests/AtlasDemoUITests.swift [hunk 3]
```diff
@@ -75,107 +71,101 @@ class AtlasDemoUITests: XCTestCase {
...
 
         tapConnectAndLogin()
-        tapBuyNow()
+        tapPlaceOrder()
         tapBackToShop()
     }
 
     func testChangeShippingAddress() {
-        let sizeText = app.tables.staticTexts["42"]
+        let size = app.cells["size-cell-1"]
```
Comment (bonkey) [inline AtlasDemo/AtlasDemoUITests/AtlasDemoUITests.swift]: Let's pull it inline into `waitForElementToAppearAndTap` call

File: AtlasDemo/AtlasDemoUITests/AtlasDemoUITests.swift [hunk 4]
```diff
@@ -75,107 +71,101 @@ class AtlasDemoUITests: XCTestCase {
...
-        tapBuyNow()
+        tapBackToSummaryFromPickingAddressButton()
+        tapPlaceOrder()
         tapBackToShop()
 
     }
 
     func testChangeBillingAddress() {
-        let sizeText = app.tables.staticTexts["42"]
+        let size = app.cells["size-cell-1"]
```
Comment (bonkey) [inline AtlasDemo/AtlasDemoUITests/AtlasDemoUITests.swift]: and again - make it inline.

**Comment by artdaw** [comment]
I wrote it in this way trying to transforming camel-case to dash-case names. I don't want to drop class names cause I want definitely to know  where the identifier is when I look at the test.

File: AtlasDemo/AtlasDemoUITests/AtlasDemoUITests.swift [hunk 5]
```diff
@@ -75,107 +71,101 @@ class AtlasDemoUITests: XCTestCase {
...
+        app.cells["address-selection-row-1"].tap()
     }
 
     private func deleteAddresses() {
-        let shippingAddressNavigationBar = app.navigationBars["Shipping Address"]
-        shippingAddressNavigationBar.buttons["Edit"].tap()
+        let rightButton = app.navigationBars.buttons["address-picker-right-button"]
+        rightButton.tap()
+
         let tablesQuery = app.tables
```
Comment (bonkey) [inline AtlasDemo/AtlasDemoUITests/AtlasDemoUITests.swift]: `let tableButtons = app.tables.buttons`

File: AtlasDemo/AtlasDemoUITests/AtlasDemoUITests.swift [hunk 6]
```diff
@@ -75,107 +71,101 @@ class AtlasDemoUITests: XCTestCase {
...
-        tablesQuery.cells.containingType(.StaticText, identifier: "Jane").staticTexts["Mollstr. 1 10178 Berlin "].tap()
-
+        app.otherElements["billing-stack-view"].tap()
+        app.cells["address-selection-row-1"].tap()
     }
 
     private func deleteAddresses() {
-        let shippingAddressNavigationBar = app.navigationBars["Shipping Address"]
-        shippingAddressNavigationBar.buttons["Edit"].tap()
+        let rightButton = app.navigationBars.buttons["address-picker-right-button"]
```
Comment (bonkey) [inline AtlasDemo/AtlasDemoUITests/AtlasDemoUITests.swift]: let's drop definition and tap instantly

**Comment by bonkey** [comment]
👍

**Comment by artdaw** [comment]
👍

**Pull request merged**